Firmware/RTK_Everywhere/Ethernet.ino

```diff
@@ -346,28 +309,61 @@ void ethernetVerifyTables()
         reportFatalError("Please fix ethernetStates table to match ethernetStatus_e");
 }
 
+void onEthernetEvent(arduino_event_id_t event, arduino_event_info_t info)
+{
+    switch (event)
+    {
+    case ARDUINO_EVENT_ETH_START:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+            systemPrintln("ETH Started");
+        // set eth hostname here
+        ETH.setHostname("esp32-eth0");
+        break;
+    case ARDUINO_EVENT_ETH_CONNECTED:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+            systemPrintln("ETH Connected");
+        break;
+    case ARDUINO_EVENT_ETH_GOT_IP:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+        {
+            systemPrintf("ETH Got IP: '%s'\n", esp_netif_get_desc(info.got_ip.esp_netif));
+            //systemPrintln(ETH);
+        }
+        eth_connected = true;
+        break;
+    case ARDUINO_EVENT_ETH_LOST_IP:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+            systemPrintln("ETH Lost IP");
+        eth_connected = false;
+        break;
+    case ARDUINO_EVENT_ETH_DISCONNECTED:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+            systemPrintln("ETH Disconnected");
+        eth_connected = false;
+        break;
+    case ARDUINO_EVENT_ETH_STOP:
+        if (settings.enablePrintEthernetDiag && (!inMainMenu))
+            systemPrintln("ETH Stopped");
+        eth_connected = false;
+        break;
+    default:
+        break;
+    }
+}
+
 //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 // Web server routines
 //=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
 
 // Start Ethernet WebServer ESP32 W5500 - needs exclusive access to WiFi, SPI and Interrupts
 void ethernetWebServerStartESP32W5500()
 {
-    // Configure the W5500
-    // To be called before ETH.begin()
-    ESP32_W5500_onEvent();
+    Network.onEvent(onEthernetEvent);
 
-    // start the ethernet connection and the server:
-    // Use DHCP dynamic IP
-    // bool begin(int POCI_GPIO, int PICO_GPIO, int SCLK_GPIO, int CS_GPIO, int INT_GPIO, int SPI_CLOCK_MHZ,
-    //            int SPI_HOST, uint8_t *W5500_Mac = W5500_Default_Mac, bool installIsrService = true);
-    ETH.begin(pin_POCI, pin_PICO, pin_SCK, pin_Ethernet_CS, pin_Ethernet_Interrupt, 25, SPI3_HOST, ethernetMACAddress);
```

Comment:
A comment here showing what the 0 and -1 arguments are for would be helpful.

Reply:
Done: 1dd9dcaec4df7dab753b89c69194a6672fed8532

```begin(ETH_PHY_TYPE, ETH_PHY_ADDR, CS, IRQ, RST, SPIClass &)```